### PR TITLE
feat(gastown): terminal fullscreen toggle and product telemetry

### PR DESCRIPTION
**Pull request by jrf0110**
## Summary

Combines three related improvements shipped together:

- **Terminal fullscreen toggle**: Double-click on the resize bar toggles the TerminalBar into fullscreen (fixed overlay, z-index 100). Escape key exits. Size is saved before entering and restored on exit. A `fullscreen` prop is forwarded to `TerminalContent` so sizing constraints are disabled. A CSS module (`TerminalBar.module.css`) provides the fullscreen and transition styles.
- **Product telemetry via PostHog**: The controller checkin now optionally includes a `productTelemetry` payload (~every 24h with ±2h jitter). The worker picks this up and sends a `kc_instance_product_telemetry` event to PostHog using a fire-and-forget `waitUntil` pattern with a 5s timeout. Skipped in development and when PostHog key is unset. Falls back to `userId` as distinct ID when Hyperdrive is unavailable.
- **Dashboard tab styling**: Refactors the three `TabsTrigger` className strings into a shared `tabTriggerClass` variable (DRY) and updates the visual style to use bordered pill tabs instead of underline tabs.
- **`getContextBalance` tRPC procedure**: New query that returns balance and depletion status for either personal or org context, with org access check.

## Verification

- Tests added for `nextProductTelemetryDeadline` (jitter bounds, edge values)
- Tests added for `collectProductTelemetry` (full config, missing file, malformed JSON, partial config, sibling-section isolation)
- Tests added for PostHog capture path in controller: absent telemetry, missing key, dev mode suppression, successful capture, Hyperdrive fallback, error resilience

## Visual Changes

N/A (tab styling change is a visual tweak but no screenshot provided — minor enough to not warrant one)

## Reviewer Notes

- The double-click detection in `onResizePointerDown` returns early on the second pointer-down event (<300ms) to prevent a drag from starting during a double-click; the actual toggle fires from the separate `onDoubleClick` handler.
- The `as Record<string, unknown>` cast in `product-telemetry.ts` follows an explicit type guard and is unavoidable at the `object` boundary.
- The PostHog project token is public by design (same as `NEXT_PUBLIC_*` convention); safe to commit.

### CODE REVIEW

File: src/components/gastown/TerminalBar.tsx [hunk 1]
```diff
@@ -146,13 +149,61 @@ export function TerminalBar({ townId, basePath: basePathOverride }: TerminalBarP
   const effectiveActiveId = activeTabId ?? 'mayor';
   const activeTab = allTabs.find(t => t.id === effectiveActiveId) ?? allTabs[0];
 
+  // ── Fullscreen state ────────────────────────────────────────────────
+  const [localFullscreen, setLocalFullscreen] = useState(false);
+  const isFullscreen = propFullscreen || localFullscreen;
```
[the remark this answers is not in the document]
Comment (jrf0110) [reply src/components/gastown/TerminalBar.tsx]: Valid concern — fixed in 3aeaf8e8. No callers pass the `fullscreen` prop (only `MayorTerminalBar` uses `TerminalBar`, passing just `townId` and `basePath`), so the prop-priority override was dead code that would have trapped the component in fullscreen if ever used.

Applied Option A: removed the `fullscreen` prop entirely and made fullscreen purely local state (`const [isFullscreen, setLocalFullscreen] = useState(false)`). Toggle via double-click and Escape key both work correctly against local state without any controlled/uncontrolled conflict.
